main.js

```diff
@@ -31,22 +30,45 @@ hlDefault.use(new defaultCodecs.MultibaseBase58btc());
  * @param {Object} options - The options for the create operation.
  * @param {Uint8Array} options.data - The data associated with the given URL. If
  *   provided, this data is used to create the cryptographic hash.
- * @param {Array} options.urls - One or more URLs that contain the data
+ * @param {Array} [options.url] - A single URL that contain the data
+ *   referred to by the hashlink.
+ * @param {Array} [options.urls] - Multiple URLs that contain the data
  *   referred to by the hashlink.
- * @param {Array} options.codecs - One or more URLs that contain the data
+ * @param {Array} [options.codecs] - One or more URLs that contain the data
  *   referred to by the hashlink.
- * @param {Object} options.meta - A set of key-value metadata that will be
+ * @param {Object} [options.meta={}] - A set of key-value metadata that will be
  *   encoded into the hashlink.
  *
  * @returns {Promise<string>} Resolves to a string that is a hashlink.
  */
-async function create({data, urls, url,
-  codecs = ['mh-sha2-256', 'mb-base58-btc'], meta = {}}) {
+async function create(options) {
+  let data, urls, url, codecs, meta;
+
+  if(!options) {
+    throw new Error('Urls and/or data params are required.');
+  }
+
+  if(typeof options === 'string') { // Convenience usage, `create(url)`
+    url = options;
+    codecs = DEFAULT_CODEC_IDS;
+    meta = {};
+  } else {
+    data = options.data;
+    urls = options.urls;
+    url = options.url;
+    codecs = options.codecs || DEFAULT_CODEC_IDS;
+    meta = options.meta || {};
+  }
 
   if(url && !urls) {
     urls = [url];
   }
 
+  if(!data) {
```

Comment:
We'll throw this elsewhere, when the resolver is called... match on scheme.

Reply:
I'd like to keep it here, if you don't mind (the error is not intelligible otherwise)